docs/docs/preparing-data.md

```diff
@@ -0,0 +1,229 @@
+                                 entityType="Country",
+                                 observationProperties={"unit": "USDollar"}
+                                 )
+```
+
+Adding data when registering a data file is optional. You can also add the data later using the `add_data` method:
+
+```python
+manager.add_data(data=df, file_name="gdp.csv")
+```
+
+[//]: # (<--- TODO: add explicit schema data file registration here --->)
+
+## Removing variables and data files
+
+You have already seen above how to remove variables and data files using the `remove_variable` 
+and `remove_data_file` methods.
+You can also remove variables and files based on their associated provenance or source.
+
+```python title="Remove variables by provenance"
+manager.remove_by_provenance("World Economic Outlook")
+```
+
+```python title="Remove data files by source"
+manager.remove_by_source("IMF")
+```
+
+
+
+## Adding and merging config files
+
+[//]: # (<---TODO: adding and merging config files here --->)
+
+
+
+
+
+## Additional settings
+
+There are some additional settings you can configure in the `CustomDataManager`:
+
+```python
+manager.set_includeInputSubdirs(True)  # Include input subdirectories in the config.json
+```
+
+This method will add the `includeInputSubdirs` field to the `config.json` file, which allows you
+to place data files in subdirectories inside the main output directory. This is useful for organizing your data files
+and keeping them structured. By default, this is not set and the Data Commons importer will expect all data files
+to be in the main output directory. If this is set to `True`, You can specify the subdirectory when adding a data file:
+
+```python
+manager.add_implicit_schema_file(file_name="economics/gdp.csv", 
+                                 ...  # other parameters as before
+                                 )
+```
+
+You can also set the `groupStatVarsByProperty` field in the `config.json` file, 
+which allows you to group statistical variables by their properties. 
+
+```python
+manager.set_groupStatVarsByProperty(True)
+```
+
+
+
+## Validating and exporting files
+
+To ensure that your `config.json` is correctly strucured, you can validate it using the `validate_config` method:
+
+```python
+manager.validate_config()
+```
+
+The config is automatically validated when you export it. You can export the `config.json` file to disk
+or get the config as a dictionary.
+
+```python title="Export config to disk"
+manager.export_config("path/to/output/config.json")
+```
+
+```python title="Get config as dictionary"
+config_dict = manager.config_to_dict()
+```
+
+Additionally, you can export the data or MCF files to disk.
+
+```python title="Export data files"
+manager.export_data("path/to/output/folder")
+```
+
+```python title="Export MCF files"
+manager.export_mfc_file("path/to/output/folder")
+```
+
+To export all files at once, you can use the `export_all` method:
+
+```python
+manager.export_all("path/to/output/folder")
```

Comment:
If you don't specify which mcf files, none are exported. This is explained in the docstring.

Reply:
Not necessarily illogical but i can explain my logic offline. It's always a question of tradeoffs with the fact that the manager can do implicit, explicit, or both